src/main/java/com/exactpro/th2/mstore/AbstractMessageProcessor.java

```diff
@@ -82,83 +83,83 @@ public AbstractMessageProcessor(
                     configuration.getDrainInterval(),
                     configuration.getDrainInterval(),
                     TimeUnit.MILLISECONDS);
-            logger.info("Drain scheduler is started");
+            LOGGER.info("Drain scheduler is started");
         }
     }
 
 
     @Override
     public void close() {
         try {
-            ScheduledFuture<?> future = this.drainFuture;
+            ScheduledFuture<?> future = drainFuture;
             if (future != null) {
                 this.drainFuture = null;
                 future.cancel(false);
             }
-        } catch (Exception ex) {
-            logger.error("Cannot cancel drain task", ex);
+        } catch (RuntimeException ex) {
+            LOGGER.error("Cannot cancel drain task", ex);
         }
 
         try {
             drain(true);
-        } catch (Exception ex) {
-            logger.error("Cannot drain left batches during shutdown", ex);
+        } catch (RuntimeException ex) {
+            LOGGER.error("Cannot drain left batches during shutdown", ex);
         }
 
         try {
             drainExecutor.shutdown();
             if (!drainExecutor.awaitTermination(configuration.getTerminationTimeout(), TimeUnit.MILLISECONDS)) {
-                logger.warn("Drain executor was not terminated during {} millis. Call force shutdown", configuration.getTerminationTimeout());
+                LOGGER.warn("Drain executor was not terminated during {} millis. Call force shutdown", configuration.getTerminationTimeout());
                 List<Runnable> leftTasks = drainExecutor.shutdownNow();
                 if (!leftTasks.isEmpty()) {
-                    logger.warn("{} tasks left in the queue", leftTasks.size());
+                    LOGGER.warn("{} tasks left in the queue", leftTasks.size());
                 }
             }
-        } catch (Exception ex) {
-            logger.error("Cannot gracefully shutdown drain executor", ex);
-            if (ex instanceof InterruptedException) {
-                Thread.currentThread().interrupt();
-            }
+        } catch (InterruptedException e) {
+            LOGGER.error("Cannot gracefully shutdown drain executor", e);
+            Thread.currentThread().interrupt();
+        } catch (RuntimeException e) {
+            LOGGER.error("Cannot gracefully shutdown drain executor", e);
         }
     }
 
     private static void confirm(Confirmation confirmation) {
         try {
             confirmation.confirm();
-        } catch (Exception e) {
-            logger.error("Exception confirming message", e);
+        } catch (IOException | RuntimeException e) {
+            LOGGER.error("Exception confirming message", e);
         }
     }
 
 
     private static void reject(Confirmation confirmation) {
         try {
             confirmation.reject();
-        } catch (Exception e) {
-            logger.error("Exception rejecting message", e);
+        } catch (IOException | RuntimeException e) {
```

Comment:
I now about hack to throw checked exception without declaration in method, but I don't think that Kotlin use it otherwise Kotlin generates unpredictable code.
Also, if Kotlin doesn't support checked exceptions that all its exception must extend RuntimeException

Reply:
Let me try to explain it a bit differently...
Kotlin throws exactly the same exceptions that Java does. But in Kotlin you don't have to declare that the method throws a checked exception. It works like SneakyThrow in Lambok...
So, there is no way in Java to know which type of exception will be thrown here... That means if we want to handle the exception here we should use `Exception` type

Reply:
Thanks for explanation, I have already changed